### DISCUSSION TITLE
live demo 是否可以禁用，如果不能，dumi-theme-mobile、自定义上下文该如何兼容

### DISCUSSION THREAD

**Discussion by onlyling** [umijs/dumi#2100] [Q&A] [answered]
在文档中没有找到如何禁用 live demo。

live demo 给 dumi-theme-mobile 带来了什么影响？

## deviceWidth 丢失

实时渲染 demo 代码导致右侧预览界面宽度抖动，不一定所有案例都撑满 deviceWidth，或少或超出。

[theme-mobile Device](https://github.com/umijs/dumi/blob/master/suites/theme-mobile/src/slots/Device/index.tsx#L20)，mobile 主题有做显示区域显示。

## 自定义上下文会不会丢失

在 `.dumi/app.tsx` 文件中，rootContainer 函数自定义了上下文。

live demo 模式会不会跳过这个函数，仅仅把 code 渲染呢？

**Comment by PeachScript** [comment, marked as answer]
暂时没有提供配置项关闭，遇到问题可以解问题

> deviceWidth 丢失

这个如何复现，官网 demo 我试了下是不会抖动的，https://d.umijs.org/guide/mobile-library#docs-guide-mobile-library-demo-0

> 自定义上下文会不会丢失

应该不会丢失，只要 Context 实例是同一份，如果有丢失的案例提供下复现

**Reply by onlyling** [reply]
deviceWidth 丢失，应该是 `dumi-theme-mobile` 没有更新到最新。
